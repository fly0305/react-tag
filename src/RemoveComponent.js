import React from 'react';
import PropTypes from 'prop-types';

const crossStr = String.fromCharCode(215);
const RemoveComponent = (props) => {
  const { readOnly, removeComponent, onClick, className } = props;
  if (readOnly) {
    return <span />;
  }

  if (removeComponent) {
    const Component = removeComponent;
    return <Component {...props} />;
  }

  return (
    <a onClick={onClick} className={className} onKeyDown={onClick}>
      {crossStr}
    </a>
  );
};

RemoveComponent.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool,
  removeComponent: PropTypes.func,
};

export default RemoveComponent;
