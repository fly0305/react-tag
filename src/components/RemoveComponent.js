import React from 'react';
import PropTypes from 'prop-types';

const crossStr = String.fromCharCode(215);
const RemoveComponent = (props) => {
  const { readOnly, removeComponent, onClick, className, tag } = props;
  if (readOnly) {
    return <span />;
  }

  if (removeComponent) {
    const Component = removeComponent;
    return <Component {...props} />;
  }

  return (
    <button onClick={onClick} className={className} aria-label={`remove ${tag.text}`}>
      {crossStr}
    </button>
  );
};

RemoveComponent.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool,
  removeComponent: PropTypes.func,
  tag: PropTypes.shape({
    id: PropTypes.string.isRequired,
    className: PropTypes.string,
    key: PropTypes.string,
  }),
};

export default RemoveComponent;
