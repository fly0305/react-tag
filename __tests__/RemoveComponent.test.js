import React from 'react';
import { shallow } from 'enzyme';
import { stub } from 'sinon';

import RemoveComponent from '../src/components/RemoveComponent';

describe('Test <RemoveComponent/>', () => {
  let minProps, onRemoveStub;
  beforeAll(() => {
    onRemoveStub = stub();
    minProps = {
      onRemove: onRemoveStub,
      index: 0,
      tag: {
        id: 'Tags',
        key: 'Tags',
      },
    };
  });
  afterAll(() => {
    onRemoveStub.restore();
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<RemoveComponent {...minProps} />);
    jestExpect(wrapper).toMatchSnapshot();
  });
});
