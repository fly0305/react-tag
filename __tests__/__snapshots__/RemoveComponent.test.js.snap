// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <RemoveComponent/> should render with base structure 1`] = `
<button
  aria-label="Tag at index 0 with value Tags focussed. Press backspace to remove"
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  Ã—
</button>
`;
